vspace/src/test/java/edu/asu/diging/vspace/core/services/impl/SpaceManagerTest.java

```diff
@@ -262,4 +263,34 @@ public void test_getSpacesWithUnPublishedStatus_spaceDoesNotExist() throws Space
         Assert.assertTrue(spaceWithNullStatus.isEmpty());
     }
 
+    @Test
+    public void test_getSpacesWithImageId_ImageIdIsValid() throws SpaceDoesNotExistException{
+        Space space=new Space();
+        space.setId(spaceId1);
+        VSImage image = new VSImage();
+        image.setId("IMG001");
+        space.setImage(image);
+        List<Space> spaceList=new ArrayList<>();
+        spaceList.add(space);
+        Mockito.when(spaceRepo.findAllByImageId("IMG001")).thenReturn(spaceList);
+        Optional<VSImage> mockImage = Optional.of(image);
+        Mockito.when(imageRepo.findById(image.getId())).thenReturn(mockImage);
+        List<ISpace> foundSpaceWithImageId= managerToTest.getSpacesWithImageId("IMG001");
+        String actualSpaceIdWithGiveImageId = foundSpaceWithImageId.get(0).getId();
+        Assert.assertEquals(spaceId1, actualSpaceIdWithGiveImageId);
+    }
+
+    @Test
+    public void test_getSpacesWithImageId_ImageIdIsInValid() throws SpaceDoesNotExistException{
+        Space space=new Space();
+        space.setId(spaceId1);
+        VSImage image = new VSImage();
+        image.setId("IMG001");
+        space.setImage(image);
+        List<Space> spaceList=new ArrayList<>();
+        Mockito.when(spaceRepo.findAllByImageId("IMG002")).thenReturn(spaceList);
+        Mockito.when(imageRepo.findById(Mockito.anyString())).thenReturn(Optional.empty());
+        Assert.assertNull(managerToTest.getSpacesWithImageId(Mockito.anyString()));
```

Comment:
`Mockito.anyString` is typically used to match arguments (in the sense of whatever string you get, do this), not as input for a method to test. Just use a string.